src/uu/cp/src/copydir.rs

```diff
@@ -213,14 +213,15 @@ where
     // `path.ends_with(".")` does not seem to work
     path.as_ref().display().to_string().ends_with("/.")
 }
-
+#[allow(clippy::too_many_arguments)]
```

Comment:
Oof. Yes, that's an okayish temporary solution.

Can I ask you to make an attempt to make a struct out of this in the next PR?

Reply:
Sure, i can make i can attempt at it, in a future PR sometime next week.  I was personally thinking about clumping the HashSet<FileInformation> types into a struct and keeping the rest of the arguments as it is . But we could discuss it in the future PR.